src/libpmem2/deep_sync_linux.c

```diff
@@ -49,3 +52,42 @@ pmem2_deep_sync_write(int region_id)
 	os_close(deep_flush_fd);
 	return 0;
 }
+
+/*
+ * pmem2_sync_cacheline -- reads file type for map and check
+ * if it is device dax or reg file, depend on file type
+ * performs proper sync operation
+ */
+int
+pmem2_sync_cacheline(struct pmem2_map *map, void *ptr, size_t size)
+{
+	enum pmem2_file_type type;
+	int ret = pmem2_get_type_from_stat(map->map_st, &type);
+	if (ret)
+		return ret;
+
+	if (type == PMEM2_FTYPE_REG) {
+		size_t len = MIN(Pagesize, size);
+		ret = pmem2_flush_file_buffers_os(map, ptr, len, 1);
```

Comment:
autorestart should be set to 0, because we can return error code
ptr should be set to the beginning of the map - ptr may not be page aligned
len should always be set to Pagesize - len may be lower than pagesize, which will probably force msync to fail

as ptr and len are not needed in this function, they can be removed